spec.bs

```diff
@@ -252,19 +252,218 @@ dictionary AuctionAdConfig {
+    to |value|, and [=iteration/continue=].
+  1. Let |buyer| be the result of [=parsing an origin=] with |key|. If |buyer| is an error,
+    [=exception/throw=] a {{TypeError}}.
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer group limits=][|buyer|] to |value|.
+1. If |config|["{{AuctionAdConfig/perBuyerExperimentGroupIds}}"] [=map/exists=], [=map/for each=]
+  |key| → |value| of |config|["{{AuctionAdConfig/perBuyerExperimentGroupIds}}"]:
+  1. If |key| equals to "*", then set |auctionConfig|'s
+    [=auction config/all buyer experiment group id=] to |value|, and [=iteration/continue=].
+  1. Let |buyer| the result of [=parsing an origin=] with |key|. If |buyer| is an error,
+    [=exception/throw=] a {{TypeError}}.
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer experiment group ids=][|buyer|] to
+    |value|.
+1. If |config|["{{AuctionAdConfig/perBuyerPrioritySignals}}"] [=map/exists=], [=map/for each=]
+  |key| → |value| of |config|["{{AuctionAdConfig/perBuyerPrioritySignals}}"]:
+  1. Let |signals| be an [=ordered map=] whose [=map/key|keys=] are [=strings=] and whose
+    [=map/value|values=] are double.
+  1. [=map/for each=] |k| → |v| of |value|:
+    1. If |k| [=string/starts with=] "browserSignals.", [=exception/throw=] a {{TypeError}}.
+    1. Set |signals|[|k|] to |v|.
+  1. If |key| equals to "*", then set |auctionConfig|'s
+    [=auction config/all buyers priority signals=] to |value|, and [=iteration/continue=].
+  1. Let |buyer| be the result of [=parsing an origin=] with |key|. If it fails, [=exception/throw=]
+    a {{TypeError}}.
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer priority signals=][|buyer|] to
+    |signals|.
+1. [=list/For each=] |component| in |config|["{{AuctionAdConfig/componentAuctions}}"]:
+  1. If |isTopLevel| is false, [=exception/throw=] a {{TypeError}}.
+  1. Let |componentAuction| be the result of running [=validate and convert auction ad config=] with
+    |component| and [=validate and convert auction ad config/isTopLevel=] set to false.
+  1. [=list/Append=] |componentAuction| to |auctionConfig|'s [=auction config/component auctions=].
+1. Return |auctionConfig|.
+
+</div>
+
+To <dfn>parse an origin</dfn> given a [=string=] |input|:
+1. Let |url| be the result of running the [=URL parser=] on |input|.
+1. If |url| is an error, then return failure.
+1. Return |url|'s [=url/origin=].
+
+To <dfn>start bidding and scoring</dfn> given an [=auction config=] |auctionConfig|:
+1. If |auctionConfig|'s [=auction config/component auctions=] are empty:
+  1. TODO: Create seller worklet.
+1. OtherWise:
+  1. [=list/For each=] |component| in |auctionConfig|'s [=auction config/component auctions=]:
+    1. [=Start bidding and scoring=] with |component|.
+  1. TODO: If the last |component|'s seller worklet is received, create seller worklet.
+1. Create bidder worklets [=in parallel=]. [=list/For each=] |buyer| in

```

Comment:
I think the name "bidder worklet" will confuse readers with the generateBid worklets; could we perhaps call this `perBuyerBidGenerator`?